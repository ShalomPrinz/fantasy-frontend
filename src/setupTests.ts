// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import React from 'react';

import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

export type TestRendererJSON = null | ReactTestRendererJSON | ReactTestRendererJSON[];
export const getJSON = (Component: React.ReactElement) => renderer.create(Component).toJSON()
