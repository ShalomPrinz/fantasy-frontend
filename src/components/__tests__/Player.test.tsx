import { getJSON, TestRendererJSON } from 'setupTests';

import { Player } from '../';

let tree: TestRendererJSON = null;

beforeEach(() => {
    tree = null
})

describe('render', () => {
    describe('with props', () => {
        it ('should render Player component', () => {
            tree = getJSON(<Player name="Shalom" team="Dortmund" width={10} widthUnits="px" />);
    
            expect(tree).toMatchSnapshot();
        })
    })
})
