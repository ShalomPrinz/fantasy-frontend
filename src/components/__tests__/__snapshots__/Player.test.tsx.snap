// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with props should render Player component 1`] = `
<div
  className="text-center"
  style={
    Object {
      "width": "14px",
    }
  }
>
  <img
    className=""
    src="Dortmund.png"
    style={
      Object {
        "width": "10px",
      }
    }
    title="Dortmund"
  />
  <div
    className="text-truncate bg-default rounded p-1 w-100"
  >
    Shalom
  </div>
</div>
`;

exports[`render with props should render full Player component 1`] = `
<div
  className="text-center"
  style={
    Object {
      "width": "14px",
    }
  }
>
  <img
    className=""
    src="Dortmund.png"
    style={
      Object {
        "width": "10px",
      }
    }
    title="Dortmund"
  />
  <div
    className="text-truncate bg-default rounded p-1 w-100"
  >
    Shalom
  </div>
</div>
`;
